### PR TITLE
Adding tests for middleware and the 3 permission possibilities 

### PR DESCRIPTION
**Pull request by Ivanca**
(Its adviced to accept the other PR first, then rebase)

### CODE REVIEW

File: seb_openedx/tests/test_middleware.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from seb_openedx.permissions import AlwaysAllowStaff, CheckSEBKeys
+
+
+class TestMiddleware(TestCase):
+    """ Tests for the seb-open-edx page """
+    def setUp(self):
+        """ setup """
+        super(TestMiddleware, self).setUp()
+        course_key_pattern = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/?]+)'
+        self.url_pattern = r'^v1/courses/{}'.format(course_key_pattern)
```
Comment (felipemontoya) [inline seb_openedx/tests/test_middleware.py:17]: Here you are using the courses API and not the course content page. It seems a little weird, but valid. Is that what you wanted, or it came from the test you took as guideline?

File: seb_openedx/tests/test_middleware.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        self.course_params = {"course_key_string": "library-v1:TestX+lib1"}
+
+    @mock.patch('seb_openedx.edxapp_wrapper.get_course_module.import_module')
+    def test_middleware_forbidden(self, m_import):
+        """ Test that middleware returns forbidden when there is no class handling allowed requests """
+        SecureExamBrowserMiddleware.allow = []
+        request = self.factory.get(self.url_pattern)
+        response = self.seb_middleware.process_view(request, self.view, [], self.course_params)
+        self.assertEqual(response.status_code, 403)
+        m_import.assert_called_with(settings.EOX_CORE_COURSE_MODULE)
```
Comment (felipemontoya) [inline seb_openedx/tests/test_middleware.py:31]: EOX_CORE? I marked this somewhere else as well.

**Comment by felipemontoya** [comment]
This looks good. I left to minor comments

File: seb_openedx/tests/test_middleware.py (hunk 2 above)
Comment (Ivanca) [reply seb_openedx/tests/test_middleware.py:31]: Will fix in next PR (to do it on this changes and existing code too)

File: seb_openedx/tests/test_middleware.py (hunk 1 above)
Comment (Ivanca) [reply seb_openedx/tests/test_middleware.py:17]: <course_key_string> is just the pattern I searched for in the codebase; but yeah its important for neither to work if user is not allowed in; maybe better to test both here (perhaps using ddt)

**Comment by Ivanca** [comment]
@felipemontoya Approved? Do I merge?

**Review by felipemontoya — APPROVED** [review]

**Pull request merged**